### PR TITLE
Automatically Parse Song Title

### PR DESCRIPTION
**Pull request by mattcheah**
In response to [issue 1:](https://github.com/NegatioN/music_consolidator/issues/1)

The URL is sent via ajax and calls youtube-dl to retrieve the title of the song, then index.jade parses it when it is returned. 

I added a loader too so that people don't try to enter their own info first which then gets overwritten by the callback.

### CODE REVIEW

File: music_consolidator/views/index.jade [hunk 1]
```diff
@@ -3,19 +3,58 @@ extends layout
...
+      function getMetaData(link) {
+        var loader = document.getElementById("loader-container");
+        loader.style.opacity = 1;
+        
+        var encodedLink = encodeURIComponent(link);
+        var request = new XMLHttpRequest();
+        
+        request.onreadystatechange = function(response) {
+          if (this.readyState == 4 && this.status == 200) {
+            var info = this.responseText.split(" - ");
```
Comment (NegatioN) [inline music_consolidator/views/index.jade:50]: Is this the responseText of the request? What's `this.readyState == 4`?

File: music_consolidator/routes/index.js [hunk 2]
```diff
@@ -34,4 +39,11 @@ function authenticate(auth_code){
     execSync(cli_call);
 }
 
+function get_metadata(link) {
+    const decoded = decodeURIComponent(link)
+    const cli_call = `youtube-dl --skip-download -e \"${decoded}\"`;
```
Comment (NegatioN) [inline music_consolidator/routes/index.js:44]: Maybe a bit clearer to use `--get-title` here instead of `-e`

**Comment by NegatioN** [comment]
Hey @mattcheah ! Really excited to see that you added this functionality, and how it turned out! :)

I have a few minor comments that either you or I could probably fix, but other then that I'm ready to take it in.

This makes me want to add support for different separators and formats as well!

File: music_consolidator/routes/index.js [hunk 3]
```diff
@@ -34,4 +39,11 @@ function authenticate(auth_code){
...
+    const child_process = require('child_process');
```
Comment (NegatioN) [inline music_consolidator/routes/index.js:45]: This should probably go at the top of the file?

**Comment by NegatioN** [comment]
Would appreciate feedback on the comments still, but I wanted to just start hacking away at it myself as soon as possible, so I merged! ;) thanks!

File: music_consolidator/views/index.jade (hunk 1 above)
Comment (mattcheah) [reply music_consolidator/views/index.jade:50]: I think readyState is set to 4 when the request is completed and returned. status == 200 checks if the response was successful or not.

**Comment by mattcheah** [comment]
Sounds good, let me know if you want me to make any of the changes or if you've got it.

**Comment by NegatioN** [comment]
@mattcheah I fixed them on master :) Thanks again for the PR!

**Comment by duraki** [comment]
Hey there, thank you both for doing this. 👍  Handy feature imo.

**Pull request merged**